Core/include/Acts/Propagator/detail/JacobianEngine.hpp

```diff
@@ -157,34 +134,47 @@ FreeToBoundMatrix freeToBoundTransportJacobian(
-/// angle
-/// @param [in] anglesToDirectionJacobian The relation jacobian from angle to
-/// dir
 /// @param [in] freeTransportJacobian Transport jacobian free to free
 /// @param [in] freeToPathDerivatives Path length derivatives for free
 ///
 /// @return a 6x8 transport jacobian from curvilinear to free
 FreeToBoundMatrix freeToCurvilinearTransportJacobian(
-    const Vector3& direction, const ActsMatrix<7, 8>& directionToAnglesJacobian,
-    const ActsMatrix<8, 7>& anglesToDirectionJacobian,
-    const FreeMatrix& freeTransportJacobian,
+    const Vector3& direction, const FreeMatrix& freeTransportJacobian,
     const FreeVector& freeToPathDerivatives);
 
-/// @brief This function calculates the free transfport jacobian from a free
-/// parameterisation.
-/// @param [in] directionToAnglesJacobian The relation jacobian from dir to
-/// angle
-/// @param [in] anglesToDirectionJacobian The relation jacobian from angle to
-/// dir
-/// @param [in] freeTransportJacobian Transport jacobian free to free
-///
-/// @return a 8x8 transport jacobian from free to free
-FreeMatrix freeToFreeTransportJacobian(
-    const ActsMatrix<7, 8>& directionToAnglesJacobian,
-    const ActsMatrix<8, 7>& anglesToDirectionJacobian,
-    const FreeMatrix& freeTransportJacobian);
+/// @brief This function reinitialises the state members required for the
+///        covariance transport
+///
+/// @param [in] geoContext The geometry context
+/// @param [in] surface The reference surface of the local parametrisation
+/// @param [in, out] freeTransportJacobian The transport jacobian from start
+///        free to final free parameters
+/// @param [in, out] freeToPathDerivatives Path length derivatives of the free,
+///        nominal parameters
+/// @param [in, out] boundToFreeJacobian Projection jacobian of the last bound
+///        parametrisation to free parameters
+/// @param [in] freeParameters Free, nominal parametrisation
+Result<void> reinitializeJacobians(const GeometryContext& geoContext,
+                                   const Surface& surface,
+                                   FreeMatrix& freeTransportJacobian,
+                                   FreeVector& freeToPathDerivatives,
+                                   BoundToFreeMatrix& boundToFreeJacobian,
+                                   const FreeVector& freeParameters);
+
+/// @brief This function reinitialises the state members required for the
+///        covariance transport
+///
+/// @param [in, out] freeTransportJacobian The transport jacobian from start
+///        free to final free parameters
+/// @param [in, out] derivatives Path length derivatives of the free, nominal
+///        parameters
+/// @param [in, out] boundToFreeJacobian Projection jacobian of the last bound
+///        parametrisation to free parameters
+/// @param [in] direction Normalised direction vector
+void reinitializeJacobians(FreeMatrix& freeTransportJacobian,
```

Comment:
Can you clarify in the comment what the difference between the two functions here is?